components/new-contribution-flow/StepProfile.js

```diff
@@ -1,43 +1,348 @@
 import PropTypes from 'prop-types';
+import { InfoCircle } from '@styled-icons/boxicons-regular/InfoCircle';
+import { defineMessages, FormattedMessage, useIntl } from 'react-intl';
 
-import { Box } from '../../components/Grid';
+import Avatar from '../../components/Avatar';
+import Container from '../../components/Container';
+import { Box, Flex } from '../../components/Grid';
+import InputTypeCountry from '../../components/InputTypeCountry';
+import Link from '../../components/Link';
+import MessageBox from '../../components/MessageBox';
+import StyledHr from '../../components/StyledHr';
+import StyledInput from '../../components/StyledInput';
+import StyledInputField from '../../components/StyledInputField';
+import StyledLink from '../../components/StyledLink';
+import StyledRadioList from '../../components/StyledRadioList';
+import StyledTextarea from '../../components/StyledTextarea';
 import { P } from '../../components/Text';
 import { withUser } from '../../components/UserProvider';
 
-class NewContributionFlowStepPayment extends React.Component {
-  static propTypes = {
-    collective: PropTypes.object,
-    LoggedInUser: PropTypes.object,
-    loadingLoggedInUser: PropTypes.object,
-    stepDetails: PropTypes.shape({
-      amount: PropTypes.number,
-      interval: PropTypes.string,
-    }),
+const messages = defineMessages({
+  incognito: { id: 'profile.incognito', defaultMessage: 'Incognito' },
+});
+
+const NewContributionFlowStepProfile = ({
+  LoggedInUser,
+  stepDetails,
+  profiles,
+  defaultSelectedProfile,
+  onChange,
+  data,
+}) => {
+  const intl = useIntl();
+  const [locale, setLocale] = useState('en');
+  const [localProfile, setLocalProfile] = useState(defaultSelectedProfile);
+
+  const renderInfoMessage = amount => {
+    if (amount < 25000) {
+      return (
+        <FormattedMessage
+          id="ContributionFlow.lowestContributionInfoMessage"
+          defaultMessage="Please provide a valid email, we need to send you a receipt for your donation."
+        />
+      );
+    } else if (amount >= 25000 && amount < 500000) {
+      return (
+        <FormattedMessage
+          id="ContributionFlow.lowContributionInfoMessage"
+          defaultMessage="Every donation must be linked to an email account for legal reasons. Please provide a valid email."
+        />
+      );
+    } else if (amount >= 500000) {
+      return (
+        <FormattedMessage
+          id="ContributionFlow.highContributionInfoMessage"
+          defaultMessage="While we'll email you a donation receipt, we are still required to keep an address for donations over $5000 USD."
+        />
+      );
+    }
   };
 
-  renderForm = () => {
-    const { LoggedInUser } = this.props;
-    const { amount, interval } = this.props.stepDetails;
-    const guestLowContribution = !LoggedInUser && amount < 500000;
-    const guestHighContribution = !LoggedInUser && amount > 500000;
+  // for guest users
+  const renderGuestForm = amount => {
```

Comment:
Same here, the form should be extracted to a separate component rather than a render func